pureMaterials/createPurematlib.py

```diff
@@ -6,73 +6,158 @@
 # -can be used for mixing homogenized regions
 # -generally impurities at <~1e-3 wt. percent (10 wppm) are removed from materials
 #   (except SS-316 steels may contain boron impurity)
-# 
+#
 #
 import os
+from typing import Dict, Union, Optional
 from pyne import material
 from pyne.material import Material, MultiMaterial
 from pyne.material_library import MaterialLibrary
+from pyne import nucname
 #
 #
 
-def make_mat(nucvec, density, citation, molecular_mass = None):
-    mat = Material(nucvec, density = density, metadata = {'citation' : citation})
+
+def enrich(
+    nucvec: Dict[int, float], old_key: int, new_values: Dict[int, float]
+) -> Dict[int, float]:
+    """Update nuclear vector by replacing an old key with new key-value pairs."""
+
+    if old_key in nucvec:
+        fract = nucvec.pop(old_key)
+        for key, value in new_values.items():
+            nucvec[key] = value * fract
+    else:
+        found_key = None
+        for key in list(nucvec.keys()):
+            if nucname.id(old_key) == nucname.id(key):
+                found_key = key
+                break
+
+        if found_key is not None:
+            fract = nucvec.pop(found_key)
+            for new_key, value in new_values.items():
+                nucvec[new_key] = value * fract  # check if percent works also
```

Comment:
Thinking of a way to streamline so the key search is separate from the substitution so that each only needs to happen once:
```
    nucvec_id_map = { nucname.id(key) : key for key in nucvec.keys() }
    old_key_id = nucname.id(old_key)

    if old_key_id in nucvec_id_map.keys():
        nucvec_key = nucvec_id_map[old_key_id]
        fract = nuvec.pop(nucvec_key)
        for nuclide, enrich_frac in new_values.items():
            nucvec[nuclide] = value * fract
```

Reply:
That is a better method, fixed one part of it here. I tested it in the test file and it gives the same output. 
`def enrich(
    nucvec: Dict[int, float], old_key: int, new_values: Dict[int, float]
) -> Dict[int, float]:
    """Update nuclear vector by replacing an old key with new key-value pairs."""

    nucvec_id_map = {nucname.id(key): key for key in nucvec.keys()}
    old_key_id = nucname.id(old_key)

    if old_key_id in nucvec_id_map.keys():
        nucvec_key = nucvec_id_map[old_key_id]
        fract = nucvec.pop(nucvec_key)
        for nuclide, enrich_frac in new_values.items():
            nucvec[nuclide] = enrich_frac * fract

    return nucvec`

Reply:
nucvec_id_map = { nucname.id(key) : key for key in nucvec.keys() }
old_key_id = nucname.id(old_key)

if old_key_id in nucvec_id_map.keys():
    nucvec_key = nucvec_id_map[old_key_id]
    fract = nuvec.pop(nucvec_key)
    for nuclide, enrich_frac in new_values.items():
        nucvec[nuclide] = enrich_frac * fract
`

Reply:
This last version has an extra loop that seems unnecessary, but maybe I'm missing something about how this works.  Isn't `old_key` necessarily an element and therefore can only appear once in `nucvec`?